src/openMVG/features/image_describer_latch.hpp

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2015 Pierre MOULON.
+
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#ifndef OPENMVG_FEATURES_LATCH_IMAGE_DESCRIBER_HPP
+#define OPENMVG_FEATURES_LATCH_IMAGE_DESCRIBER_HPP
+
+#include <iostream>
+#include <numeric>
+
+#include "openMVG/features/image_describer.hpp"
+#include "openMVG/features/regions_factory.hpp"
+#include <cereal/cereal.hpp>
+
+#ifdef OpenMVG_USE_CUDA
+#include "third_party/koral/KORAL.h"
+#endif
+
+#ifdef OPENMVG_USE_SSE
+#include <xmmintrin.h>
+#endif
+
+#ifdef OpenMVG_USE_CUDA
+
+using namespace std;
+
+namespace openMVG {
+namespace features {
+
+enum ELATCH_DESCRIPTOR
+{
+  LATCH_BINARY
+};
+
+struct LATCHParams
+{
+  LATCHParams(
+    ELATCH_DESCRIPTOR eLatchDescriptor = LATCH_BINARY
+  ):eLatchDescriptor_(eLatchDescriptor){}
+
+  template<class Archive>
+  void serialize(Archive & ar)
+  {
+    ar(eLatchDescriptor_);
+  }
+
+  // Parameters
+  ELATCH_DESCRIPTOR eLatchDescriptor_;
+};
+
+class LATCH_Image_describer : public Image_describer
+{
+public:
+  LATCH_Image_describer(
+  const LATCHParams & params = LATCHParams()
+  ):Image_describer(), params_(params), koral_(1.2f, 8) {
+        }
+
+  // Don't need to really define this for the LATCH class yet, until more descriptors come out.
+  bool Set_configuration_preset(EDESCRIBER_PRESET preset) override
+  {
+    return true;
+  }
+
+  /**
+  @brief Detect regions on the image and compute their attributes (description)
+  @param regions The detected regions and attributes (the caller must delete the allocated data)
+  @param mask 8-bit gray image for keypoint filtering (optional).
+     Non-zero values depict the region of interest.
+  */
+  bool Describe(const image::Image<unsigned char>& image,
+    std::unique_ptr<Regions> &regions,
+    const image::Image<unsigned char> * mask = nullptr ) override
+  {
+    std::vector<Keypoint> kpts;
+    std::vector<uint64_t> descriptors;
+
+    constexpr uint8_t KFAST_thresh = 60;
+    koral_.go(image.data(), image.cols(), image.rows(), KFAST_thresh);
+    kpts = koral_.kps;
```

Comment:
Seems like those copy are not necessary, since we can use directly the `koral_.kps` and `koral_.desc`.
If you want use short name you can use some `const auto&` alias.